### DISCUSSION TITLE
how to add spliced/unspliced information to my h5ad object

### DISCUSSION THREAD

**Discussion by FADHLyemen** [theislab/scvelo#494] [Q&A] [answered]
Hi, 
I want to test scvelo but do not have the splice/unsplice in my object in layer, how to add it to my object?

Thank you

**Comment by WeilerP** [comment]
@FADHLyemen, you can do the following

```python
adata.layers['unspliced'] = unspliced_array
adata.layers['spliced'] = spliced_array
```

where `unspliced_array`, `spliced_array` are `numpy.ndarray` or sparse matrices.

**Comment by FADHLyemen** [comment]
Thank you, but how get spliced_array, is it output from cell ranger?

How it look likes? is it a count such X in my h5ad?

Thank you

**Comment by WeilerP** [comment, marked as answer]
@FADHLyemen you need to run the [velocyto](http://velocyto.org/velocyto.py/tutorial/index.html#running-the-cli) pipeline, then read the LOOM file.

In general: Please reply directly to a suggested answer instead of opening a new one which is simply a response to a suggested one.

**Reply by FADHLyemen** [reply]
@WeilerP 
I need to apply scvelo on my data, I have one question on how to get the splice and un-splice information for cells. can I extract it from scRNA-seq or I should have ATAC-seq data to get splice and un-splice  to run the velocity program?

**Reply by WeilerP** [reply]
There is not need for ATAC-seq data. You only need scRNA-seq data.
